src/main/java/org/example/hospital/convertors/DoctorConverter.java

```diff
@@ -0,0 +1,31 @@
+package org.example.hospital.convertors;
+
+import org.example.hospital.dto.DoctorDTO;
+import org.example.hospital.entity.Doctor;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DoctorConverter implements Converter<Doctor, DoctorDTO>{
+
+    @Override
+    public DoctorDTO convertToDTO(Doctor entity, DoctorDTO dto) {
+        dto.setDoctorId(entity.getDoctorId());
+        dto.setName(entity.getName());
+        dto.setRoomNumber(entity.getRoomNumber());
+        dto.setPhoneNumber(entity.getPhoneNumber());
+        dto.setInfo(entity.getInfo());
+        return dto;
+    }
+
+    @Override
+    public Doctor convertToEntity(DoctorDTO dto, Doctor entity) {
+        if (entity.getDoctorId() == null) {
```

Comment:
Please remove null checking here too

Reply:
There is no need to set an Id during an update (Id will be overwritten and cause an error), so I have a null checker to set Id only for new entities.